src/config.rs

```diff
@@ -194,4 +243,13 @@ mod test {
             Setting::String("wont accept strings".into()),
         );
     }
+
+    #[test]
+    fn find_settings() {
+        let enabled = config!(bool "dns.local_resolver.enabled");
```

Comment:
this is ultimately what i want to do:   have a single macro that allows us to fetch a value from the config store, whereever it may be..   Since it's the user-agent that initializes the config store, we must have some global place where this information can be found.. we can't init our own config store within the engine.

We therefor must have a place where the engine always can find the store.. it sounds like a singleton (not a big fan though).. but we can't pass a config store to each deep function just in case we need it somewhere.